library/src/scripts/components/SelectBox.tsx

```diff
@@ -0,0 +1,140 @@
+import DropDown from "@library/components/dropdown/DropDown";
+import { ButtonBaseClass } from "@library/components/forms/Button";
+
+export interface ISelectBoxItem {
+    name: string;
+    className?: string;
+    onClick: () => {};
+    selected?: boolean;
+    outdated?: boolean;
+}
+
+interface IProps {
+    className?: string;
+    id?: string;
+    children: ISelectBoxItem[];
+    stickTop?: boolean; // Adjusts the flyout position vertically
+    stickRight?: boolean; // Adjusts the flyout position horizontally
+}
+
+export interface ISelfLabelledProps extends IProps {
+    label: string;
+}
+
+export interface IExternalLabelledProps extends IProps {
+    describedBy: string;
+}
+
+interface IState {
+    id: string;
+    selectedIndex: number;
+    selectedItem: any;
+}
+
+/**
+ * Generates Select Box component (similar to a select)
+ */
+export default class SelectBox extends React.Component<ISelfLabelledProps | IExternalLabelledProps, IState> {
+    public static defaultProps = {
+        selectedIndex: 0,
+    };
+
+    public constructor(props) {
+        super(props);
+
+        this.state = {
+            id: getRequiredID(props, "selectBox-"),
+            selectedIndex: props.selectedIndex,
+            selectedItem: props.selectedItem || props.children[props.selectedIndex],
+        };
+    }
+
+    /**
+     * Handle click on item in select box.
+     * @param selectedItem data for item
+     * @param index the index of the item
+     */
+    public handleClick = (selectedItem: ISelectBoxItem, index: number) => {
+        this.setState({
+            selectedIndex: index,
+            selectedItem,
+        });
+    };
+
+    public render() {
+        const space = `&nbsp;`;
+
+        const selectItems =
+            this.props.children && this.props.children.length > 0
+                ? this.props.children.map((child, i) => {
+                      const selected = this.state.selectedIndex === i;
+                      return (
+                          <DropDownItemButton
+                              key={this.props.id + "-item" + i}
+                              className={classNames({ isSelected: child.selected })}
+                              name={child.name}
+                              onClick={this.handleClick.bind(this, child, i)}
+                              disabled={i === this.state.selectedIndex}
+                              clickData={child}
+                              index={i}
+                              current={selected}
+                              buttonClassName="dropDownItem-button selectBox-buttonItem"
+                          >
+                              <span className="selectBox-checkContainer sc-only">
+                                  {selected && checkCompact("selectBox-selectedIcon")}
+                                  {!selected && (
+                                      <span className="selectBox-spacer" dangerouslySetInnerHTML={{ __html: space }} />
```

Comment:
This is not the correct way to set a space inside of the element. We should almost _never_ use dangerouslySetInnerHTML.
Here `<span className="selectBox-spacer">{` `}</span>}` will work. Gave it a quick test to double check. You can't specify the encoded form of HTML entities though. You should specify the plain form, and react will handle converting to HTML entities for you.